[Bug]: package/feeds/passwall_packages/gn [host] failed to build.

### 描述您遇到的bug

gn: update to git HEAD #114
此合并请求提交以后，再也没有编译成功

### 复现此Bug的步骤

使用github的Actions编译lede-openwrt
自用脚本https://github.com/justlovemaki/OpenWRT-CI

### 您想要实现的目的

修复此问题

### 日志信息

2024-02-06T23:10:51.6238293Z make[3]: Entering directory '/home/runner/work/OpenWRT-CI/OpenWRT-CI/owrt/feeds/passwall_packages/gn'
2024-02-06T23:10:51.7227183Z MAKEFLAGS="" /home/runner/work/OpenWRT-CI/OpenWRT-CI/owrt/staging_dir/host/bin/ninja  -j1 -C /home/runner/work/OpenWRT-CI/OpenWRT-CI/owrt/build_dir/hostpkg/gn-2024-01-24-a2e2717e/out
2024-02-06T23:10:51.7260356Z ninja: Entering directory `/home/runner/work/OpenWRT-CI/OpenWRT-CI/owrt/build_dir/hostpkg/gn-2024-01-24-a2e2717e/out'
2024-02-06T23:10:51.7310363Z [1/287] CXX src/base/command_line.o
2024-02-06T23:10:51.7310936Z FAILED: src/base/command_line.o 
2024-02-06T23:10:51.7314531Z clang++ -MMD -MF src/base/command_line.o.d -I../src -I. -DNDEBUG -O3 -fdata-sections -ffunction-sections -Werror -D_FILE_OFFSET_BITS=64 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -pthread -pipe -fno-exceptions -fno-rtti -fdiagnostics-color -Wall -Wextra -Wno-unused-parameter -Wextra-semi -Wundef -std=c++20 -Wrange-loop-analysis -Wextra-semi-stmt -c ../src/base/command_line.cc -o src/base/command_line.o
2024-02-06T23:10:51.7317386Z make[3]: *** [Makefile:58: /home/runner/work/OpenWRT-CI/OpenWRT-CI/owrt/build_dir/hostpkg/gn-2024-01-24-a2e2717e/.built] Error 1
2024-02-06T23:10:51.7318138Z /bin/sh: 1: clang++: not found
2024-02-06T23:10:51.7318469Z ninja: build stopped: subcommand failed.
2024-02-06T23:10:51.7319186Z make[3]: Leaving directory '/home/runner/work/OpenWRT-CI/OpenWRT-CI/owrt/feeds/passwall_packages/gn'
2024-02-06T23:10:51.7320281Z time: package/feeds/passwall_packages/gn/host-compile#0.09#0.03#0.11
2024-02-06T23:10:51.7328548Z     ERROR: package/feeds/passwall_packages/gn [host] failed to build.
2024-02-06T23:10:51.7333271Z make[2]: *** [package/Makefile:114: package/feeds/passwall_packages/gn/host/compile] Error 1
2024-02-06T23:10:51.7337804Z make[2]: Leaving directory '/home/runner/work/OpenWRT-CI/OpenWRT-CI/owrt'
2024-02-06T23:10:51.7342818Z make[1]: *** [package/Makefile:110: /home/runner/work/OpenWRT-CI/OpenWRT-CI/owrt/staging_dir/target-x86_64_musl/stamp/.package_compile] Error 2
2024-02-06T23:10:51.7347685Z make[1]: Leaving directory '/home/runner/work/OpenWRT-CI/OpenWRT-CI/owrt'
2024-02-06T23:10:51.7352967Z make: *** [/home/runner/work/OpenWRT-CI/OpenWRT-CI/owrt/include/toplevel.mk:231: world] Error 2
2024-02-06T23:10:51.7462364Z ##[error]Process completed with exit code 2.

### 截图

_No response_

### 系统相关信息

github的Actions

### 其他信息

_No response_

--- a/gn/patches/010-gcc.patch
+++ b/gn/patches/010-gcc.patch
@@ -1,0 +1,11 @@
+--- a/build/gen.py
++++ b/build/gen.py
+@@ -368,7 +368,7 @@ def WriteGNNinja(path, platform, host, o
+     cxx = os.environ.get('CXX', 'g++')
+     ld = os.environ.get('LD', 'g++')
+     ar = os.environ.get('AR', 'ar -X64')
+-  elif platform.is_msys() or platform.is_mingw():
++  elif platform.is_msys() or platform.is_mingw() or platform.is_linux():
+     cxx = os.environ.get('CXX', 'g++')
+     ld = os.environ.get('LD', 'g++')
+     ar = os.environ.get('AR', 'ar')

